MvvmCross-iOSSupport/MvvmCross.iOS.Support.XamarinSidebar/MvxSidebarPresenter.cs

```diff
@@ -33,94 +32,77 @@ protected virtual bool PresentationHintHandler(MvxPanelPresentationHint hint)
+
+                    ChangePresentation(new MvxSidebarResetRootPresentationHint(attribute.Panel, SideBarViewController, viewController));
                     break;
                 case MvxPanelHintType.ActivePanel:
                 default:
-                    ChangePresentation(new MvxSidebarActivePanelPresentationHint(viewPresentationAttribute.Panel, RootViewController, viewController));
+
+                    ChangePresentation(new MvxSidebarActivePanelPresentationHint(attribute.Panel, SideBarViewController, viewController));
                     break;
             }
 
-            if (!viewPresentationAttribute.ShowPanel)
+            if (!attribute.ShowPanel)
             {
-                var menu = Mvx.Resolve<IMvxSideMenu>();
-                menu?.Close();
+                var menu = Mvx.Resolve<IMvxSidebarViewController>();
+                menu?.CloseMenu();
             }
         }
 
-        public override void Close(IMvxViewModel toClose)
+        protected override void ShowRootViewController(UIViewController viewController, MvxRootPresentationAttribute attribute, MvxViewModelRequest request)
         {
-            if (ParentRootViewController.ViewControllers.Count() > 1)
-                ParentRootViewController.PopViewController(true);
-            else if (RootViewController.NavigationController.ViewControllers.Count() > 1)
-                RootViewController.NavigationController.PopViewController(true);
-            else
-                base.Close(toClose);
-        }
+            // check if viewController is a MvxSidebarPanelController
+            if (viewController is IMvxSidebarViewController)
+            {
+                MasterNavigationController = new MvxNavigationController();
 
-        protected MvxPanelPresentationAttribute GetViewPresentationAttribute(IMvxIosView view)
-        {
-            if (view == null)
-                return default(MvxPanelPresentationAttribute);
+                SideBarViewController = viewController as IMvxSidebarViewController;
+                SideBarViewController.SetNavigationController(MasterNavigationController);
 
-            return view.GetType().GetCustomAttributes(typeof(MvxPanelPresentationAttribute), true).FirstOrDefault() as MvxPanelPresentationAttribute;
-        }
+                SetWindowRootViewController(viewController);
 
-        protected virtual void InitRootViewController()
-        {
-            foreach (var view in Window.Subviews)
-                view.RemoveFromSuperview();
+                Mvx.RegisterSingleton<IMvxSidebarViewController>(SideBarViewController);
+
+                CloseMasterNavigationController();
+                CloseModalNavigationController();
+                CloseSplitViewController();
 
-            MasterNavigationController = new UINavigationController();
+                return;
+            }
 
```

Comment:
I think there should be an `else` here that releases all references to `IMvxSidebarViewController`, since if the upper `if` is not true, it means the user is changing the root of the app